src/promptflow-devkit/promptflow/_proxy/_python_inspector_proxy.py

```diff
@@ -35,7 +36,10 @@ def get_entry_meta(
         **kwargs,
     ) -> Dict[str, Any]:
         timeout = kwargs.get("timeout", FLOW_META_JSON_GEN_TIMEOUT)
-        load_in_subprocess = kwargs.get("load_in_subprocess", True)
+        load_in_subprocess = (
+            kwargs.get("load_in_subprocess", True)
+            and os.environ.get(PF_FLOW_META_LOAD_IN_SUBPROCESS, "True").lower() == "true"
```

Comment:
It's quite not a common practice to check environment variable to determine a util method's behavior.
Maybe we could move this check operation to non-util method?
Not sure which code path you need to cover.
![image](https://github.com/microsoft/promptflow/assets/17527303/30a8ccd4-a5b7-481d-824b-afd17a2ba290)

Reply:
The purpose of this environment variable is to offer a way to override the default flow meta loading behavior in the `get_entry_meta` through an environment variable. 

Since `get_entry_meta` is the initial function that introduces the `load_in_subprocess`, I think it’s an appropriate location for this override check. Moving this check to upper layer functions would require method signature changes in multiple places, also need to accommodate different code paths. Current approach tries to maintain code simplicity while fulfilling the needs of the change. Hope this makes sense.